### PR TITLE
Fix create user error

### LINKED ISSUE #1677: Issue creating new users
When attempting to create a new user there is an error with message: "UserRegistrationError: email, password and retyped password are required!"

https://github.com/adaptlearning/adapt_authoring/pull/1551 introduced this issue.

### PR DESCRIPTION
**Pull request by canstudios-nicolaw**
Fixes #1677

### CODE REVIEW

File: plugins/auth/local/index.js [hunk 1]
```diff
@@ -195,13 +195,19 @@ LocalAuth.prototype.registerUser = function (req, res, next) {
 
 };
 
-LocalAuth.prototype.internalRegisterUser = function (user, cb) {
-  if (!user.email || !user.password || !user.retypePassword) {
+LocalAuth.prototype.internalRegisterUser = function(retypePasswordRequired, user, cb) {
+  if (!user.email || !user.password) {
     return cb(new auth.errors.UserRegistrationError('email, password and retyped password are required!'));
```
Comment (tomgreenfield) [inline plugins/auth/local/index.js]: Just to note this error message could now be misleading when there's no retyped password field:

retyped

Comment (canstudios-nicolaw) [inline plugins/auth/local/index.js]: Good spot I'll fix that now

File: plugins/auth/local/index.js [hunk 2]
```diff
@@ -195,13 +195,19 @@ LocalAuth.prototype.registerUser = function (req, res, next) {
...
+    if (!user.email || !user.password || !user.retypePassword) {
+      return cb(new auth.errors.UserRegistrationError('email, password and retyped password are required!'));
+    }
 
-  if (user.password !== user.retypePassword) {
-    return cb(new auth.errors.UserRegistrationError('password and retyped password must match!'))
+    if (user.password !== user.retypePassword) {
+      return cb(new auth.errors.UserRegistrationError('password and retyped password must match!'));
+    }
+  } else {
```
Comment (tomgreenfield) [inline plugins/auth/local/index.js]: Cheers for amending. Can we lose the `else` here, unless I'm missing something?

Comment (canstudios-nicolaw) [inline plugins/auth/local/index.js]: Yes we can! 2 ticks

**Pull request merged**